### PR TITLE
board: add support for VCC-GND STM32H750VB

### PR DESCRIPTION
**Pull request by emhlb**
YD-STM23H750VB board has 128 Kbytes of flash memory, 1 Mbytes of SRAM and 16 Mbytes onboard SPI norflash.

### CODE REVIEW

File: boards/vcc-gnd/yd_stm32h750vb/yd_stm32h750vb.dts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+&usart1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&usart1_tx_pa9 &usart1_rx_pa10>;
+	current-speed = <115200>;
+	status = "okay";
+};
+
+&sdmmc1 {
+	pinctrl-names = "default";
```
Comment (erwango) [inline boards/vcc-gnd/yd_stm32h750vb/yd_stm32h750vb.dts:101]: status = "okay";

File: boards/vcc-gnd/yd_stm32h750vb/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+Flashing an application to YD-STM32H750VB
+-----------------------------------------
+
+Here is an example for the :zephyr:code-sample:`blinky` application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/basic/blinky
+   :board: yd_stm32h750vb
+   :goals: build flash
```
Comment (erwango) [inline boards/vcc-gnd/yd_stm32h750vb/doc/index.rst:77]: This part is a little bit vague. 
How does it relate with openocd and jlink runners configured ?

File: boards/vcc-gnd/yd_stm32h750vb/yd_stm32h750vb.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+arch: arm
+toolchain:
+  - zephyr
+  - gnuarmemb
+  - xtools
+ram: 1024
+flash: 128
+supported:
+  - gpio
+  - dma
```
Comment (erwango) [inline boards/vcc-gnd/yd_stm32h750vb/yd_stm32h750vb.yaml:13]: This could be reviewed I guess

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for contributing this board.

File: boards/vcc-gnd/yd_stm32h750vb/doc/index.rst (hunk 2 above)
Comment (JohnSanpe) [reply boards/vcc-gnd/yd_stm32h750vb/doc/index.rst]: Added some descriptions about using BOOT0 DFU mode

File: boards/vcc-gnd/yd_stm32h750vb/yd_stm32h750vb.yaml (hunk 3 above)
Comment (JohnSanpe) [reply boards/vcc-gnd/yd_stm32h750vb/yd_stm32h750vb.yaml:13]: Do we need more information to describe here？

Comment (erwango) [reply boards/vcc-gnd/yd_stm32h750vb/yd_stm32h750vb.yaml:13]: Idea is to mention what are the devices/nodes enabled on the board

**Comment by JohnSanpe** [comment]
The above issue has been resolved.

**Review by erwango — DISMISSED** [review]

File: boards/vcc-gnd/yd_stm32h750vb/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,137 @@
+.. _yd_stm32h750vb:
+
+YD-STM32H750VB
+##############
+
+Overview
+********
+
+The YD-STM32H750VB development board is a complete demonstration and development
+platform for Arm® Cortex®-M7 core-based STM32H750VBT6 microcontroller, with
```
Comment (ajarmouni-st) [inline boards/vcc-gnd/yd_stm32h750vb/doc/index.rst:10]:
```suggestion
platform for Arm |reg| Cortex |reg|-M7 core-based STM32H750VBT6 microcontroller, with
```

File: boards/vcc-gnd/yd_stm32h750vb/doc/index.rst [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port-polling;                |
+|           |            | serial port-interrupt               |
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| RTC       | on-chip    | Real Time Clock                     |
```
Comment (ajarmouni-st) [inline boards/vcc-gnd/yd_stm32h750vb/doc/index.rst:42]: Not enabled in .dts, should be removed from doc.

File: boards/vcc-gnd/yd_stm32h750vb/yd_stm32h750vb.yaml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  - gnuarmemb
+  - xtools
+ram: 1024
+flash: 128
+supported:
+  - gpio
+  - uart
+  - qspi
+  - sdhc
+  - rtc
```
Comment (ajarmouni-st) [inline boards/vcc-gnd/yd_stm32h750vb/yd_stm32h750vb.yaml:16]: Not enabled in .dts, should be removed.

File: boards/vcc-gnd/yd_stm32h750vb/yd_stm32h750vb.yaml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+identifier: yd_stm32h750vb
+name: YD-STM32H750VB
+type: mcu
+arch: arm
+toolchain:
+  - zephyr
+  - gnuarmemb
+  - xtools
```
Comment (ajarmouni-st) [inline boards/vcc-gnd/yd_stm32h750vb/yd_stm32h750vb.yaml:8]: Have you tested these toolchains?

File: boards/vcc-gnd/yd_stm32h750vb/yd_stm32h750vb.yaml [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+ram: 1024
+flash: 128
+supported:
+  - gpio
+  - uart
+  - qspi
+  - sdhc
```
Comment (ajarmouni-st) [inline boards/vcc-gnd/yd_stm32h750vb/yd_stm32h750vb.yaml:15]: Not needed, see https://github.com/zephyrproject-rtos/zephyr/pull/72695

File: boards/vcc-gnd/yd_stm32h750vb/yd_stm32h750vb.yaml [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+  - rtc
+  - usb
+  - usb_device
```
Comment (ajarmouni-st) [inline boards/vcc-gnd/yd_stm32h750vb/yd_stm32h750vb.yaml:18]: Have you run USB tests?

File: boards/vcc-gnd/yd_stm32h750vb/board.cmake [hunk 10]
```diff
@@ -0,0 +1,9 @@
+# SPDX-License-Identifier: Apache-2.0
+
+board_runner_args(dfu-util "--pid=0483:df11" "--alt=0" "--dfuse")
+board_runner_args(jlink "--device=STM32H735IG" "--speed=4000")
+board_runner_args(openocd --target-handle=_CHIPNAME.cpu0)
+
+include(${ZEPHYR_BASE}/boards/common/dfu-util.board.cmake)
```
Comment (ajarmouni-st) [inline boards/vcc-gnd/yd_stm32h750vb/board.cmake:7]: Could you test if STM32CubeProgrammer works & add it?
```suggestion
board_runner_args(openocd --target-handle=_CHIPNAME.cpu0)
board_runner_args(stm32cubeprogrammer "--port=swd" "--reset-mode=hw")

include(${ZEPHYR_BASE}/boards/common/dfu-util.board.cmake)
include(${ZEPHYR_BASE}/boards/common/stm32cubeprogrammer.board.cmake)
```

**Review by ajarmouni-st — CHANGES_REQUESTED** [review]

File: boards/vcc-gnd/yd_stm32h750vb/yd_stm32h750vb.yaml (hunk 9 above)
Comment (JohnSanpe) [reply boards/vcc-gnd/yd_stm32h750vb/yd_stm32h750vb.yaml]: Does the USB driver seem to have some issues? I tested UDC and it failed, so I decided to delete it directly

```
*** Booting Zephyr OS build v3.6.0-1948-g9965ed17614d ***
Running TESTSUITE udc_driver_test
===================================================================
I: UDC device HS: 0
START - test_udc_ep_buf
E: Unable to allocate FIFO for 0x81

    Assertion failed at WEST_TOPDIR/zephyr/tests/drivers/udc/src/main.c:332: test_udc_ep_api: (err is non-zero)
Failed to enable endpoint
 FAIL - test_udc_ep_buf in 0.083 seconds
===================================================================
START - test_udc_ep_int

    Assertion failed at WEST_TOPDIR/zephyr/tests/drivers/udc/src/main.c:377: test_udc_ep_mps: (err is non-zero)
Failed to initialize UDC driver
 FAIL - test_udc_ep_int in 0.013 seconds
===================================================================
START - test_udc_ep_iso

    Assertion failed at WEST_TOPDIR/zephyr/tests/drivers/udc/src/main.c:377: test_udc_ep_mps: (err is non-zero)
Failed to initialize UDC driver
 FAIL - test_udc_ep_iso in 0.013 seconds
===================================================================
TESTSUITE udc_driver_test failed.

------ TESTSUITE SUMMARY START ------

SUITE FAIL -   0.00% [udc_driver_test]: pass = 0, fail = 3, skip = 0, total = 3 duration = 0.109 seconds
 - FAIL - [udc_driver_test.test_udc_ep_buf] duration = 0.083 seconds
 - FAIL - [udc_driver_test.test_udc_ep_int] duration = 0.013 seconds
 - FAIL - [udc_driver_test.test_udc_ep_iso] duration = 0.013 seconds

------ TESTSUITE SUMMARY END ------

===================================================================
PROJECT EXECUTION FAILED
```

Comment (JohnSanpe) [reply boards/vcc-gnd/yd_stm32h750vb/yd_stm32h750vb.yaml]: By the way, the STM32CubeProgrammer is working properly.

```
      -------------------------------------------------------------------
                        STM32CubeProgrammer v2.16.0                  
      -------------------------------------------------------------------

ST-LINK SN  : 0F1A090029135147324D4E00
ST-LINK FW  : V2J43S7
Board       : --
Voltage     : 3.02V
SWD freq    : 4000 KHz
Connect mode: Under Reset
Reset mode  : Hardware reset
Device ID   : 0x450
Revision ID : Rev V
Device name : STM32H7xx
Flash size  : 128 KBytes
Start Address : 8000000
Device type : MCU
Device CPU  : Cortex-M7
BL Version  : 0x91



Memory Programming ...
Opening and parsing file: zephyr.hex
  File          : zephyr.hex
  Size          : 57.55 KB 
  Address       : 0x08000000 


Erasing memory corresponding to segment 0:
Download in Progress:
[==================================================] 100% 

File download complete
Time elapsed during download operation: 00:00:13.438

RUNNING Program ... 
  Address:      : 0x8000000
Application is running, Please Hold on...
Start operation achieved successfully
```

File: boards/vcc-gnd/yd_stm32h750vb/yd_stm32h750vb.yaml (hunk 7 above)
Comment (JohnSanpe) [reply boards/vcc-gnd/yd_stm32h750vb/yd_stm32h750vb.yaml:8]: There's no problem with these

**Review by ajarmouni-st — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Comment by kartben** [comment]
Ping @erwango

File: boards/vcc-gnd/yd_stm32h750vb/yd_stm32h750vb_defconfig [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+CONFIG_UART_CONSOLE=y
+
+# enable GPIO
+CONFIG_GPIO=y
+
+# Enable Clocks
+CONFIG_CLOCK_CONTROL=y
+
+# enable pin controller
+CONFIG_PINCTRL=y
```
Comment (erwango) [inline boards/vcc-gnd/yd_stm32h750vb/yd_stm32h750vb_defconfig:26]: Non blocking: Could be removed.

**Review by erwango — APPROVED** [review]

**Pull request merged**